File: tools/backfill_updated.py
#!/usr/bin/env python3
"""updated 필드가 없는 .md 파일에 마지막 git 커밋 날짜를 삽입"""
import re
import subprocess
from pathlib import Path

REPO_ROOT = Path(__file__).parent.parent
DOCS_DIR = REPO_ROOT / "Develop"
FRONT_MATTER_RE = re.compile(r"^---\s*\n(.*?)\n---", re.DOTALL)

fixed = 0
for md in sorted(DOCS_DIR.rglob("*.md")):
    content = md.read_text(encoding="utf-8")
    m = FRONT_MATTER_RE.match(content)
    if not m:
        continue
    fm = m.group(1)
    if re.search(r"^updated\s*:", fm, re.MULTILINE):
        continue

    # git log date
    result = subprocess.run(
        ["git", "log", "-1", "--format=%ad", "--date=format:%Y-%m-%d", "--", str(md)],
        capture_output=True, text=True, cwd=REPO_ROOT
    )
    git_date = result.stdout.strip() or "2026-08-07"

    # 두 번째 --- 직전에 삽입
    close = m.end()
    insert_pos = content.rfind("\n---", m.start(), close) + 1
    new_content = content[:insert_pos] + f"updated: {git_date}\n" + content[insert_pos:]
    md.write_text(new_content, encoding="utf-8")
    fixed += 1

print(f"완료: {fixed}개 파일 updated 백필")

File: tools/backfill_updated.mjs
#!/usr/bin/env node
import { execSync } from 'child_process';
import { readFileSync, writeFileSync, readdirSync, statSync } from 'fs';
import { join, relative } from 'path';

const REPO_ROOT = new URL('..', import.meta.url).pathname.replace(/\/$/, '');
const DOCS_DIR = join(REPO_ROOT, 'Develop');
const FRONT_RE = /^---\s*\n([\s\S]*?)\n---/;

function findMd(dir) {
  const result = [];
  for (const entry of readdirSync(dir, { withFileTypes: true })) {
    const full = join(dir, entry.name);
    if (entry.isDirectory()) result.push(...findMd(full));
    else if (entry.isFile() && entry.name.endsWith('.md')) result.push(full);
  }
  return result;
}

let fixed = 0;
for (const md of findMd(DOCS_DIR).sort()) {
  const content = readFileSync(md, 'utf8');
  const m = content.match(FRONT_RE);
  if (!m) continue;
  if (/^updated\s*:/m.test(m[1])) continue;

  const gitDate = execSync(
    `git log -1 --format=%ad --date=format:%Y-%m-%d -- "${relative(REPO_ROOT, md)}"`,
    { cwd: REPO_ROOT, encoding: 'utf8' }
  ).trim() || '2026-08-07';

  // 두 번째 --- 직전에 삽입
  const closeIdx = content.indexOf('\n---', m.index + 4);
  const newContent = content.slice(0, closeIdx + 1) + `updated: ${gitDate}\n` + content.slice(closeIdx + 1);
  writeFileSync(md, newContent, 'utf8');
  fixed++;
}

console.log(`완료: ${fixed}개 파일 updated 백필`);
